src/subdomains/supporting/support-issue/services/support-issue.service.ts

```diff
@@ -27,17 +31,53 @@ export class SupportIssueService {
 
     const user = await this.userService.getUser(userId, { userData: true });
 
-    const entity = this.supportIssueRepo.create({ type: SupportIssueType.TRANSACTION_ISSUE, ...dto });
+    let entity = this.supportIssueRepo.create({ type: SupportIssueType.TRANSACTION_ISSUE, ...dto });
 
     entity.transaction = await this.transactionService.getTransactionById(transactionId, { user: { userData: true } });
     if (!entity.transaction) throw new NotFoundException('Transaction not found');
     if (!entity.transaction.user || entity.transaction.user.userData.id !== user.userData.id)
       throw new ForbiddenException('You can only create support issue for your own transaction');
 
+    entity = await this.supportIssueRepo.save(entity);
+
+    if (dto.message)
+      await this.createSupportMessage(
+        {
+          author: SupportMessageAuthor.CUSTOMER,
+          supportIssue: entity,
+          ...dto,
+        },
+        userId,
+      );
+  }
+
+  async updateSupportIssue(id: number, dto: UpdateSupportIssueDto): Promise<SupportIssue> {
+    const entity = await this.supportIssueRepo.findOneBy({ id });
+    if (!entity) throw new NotFoundException('SupportIssue not found');
+
+    Object.assign(entity, dto);
+
+    return this.supportIssueRepo.save(entity);
+  }
+
+  async createSupportMessage(dto: CreateSupportMessageDto, userId?: number): Promise<void> {
+    const existing = await this.messageRepo.findOneBy({
+      message: dto.message,
+      supportIssue: { id: dto.supportIssue.id },
+    });
+    if (existing) throw new ConflictException('Support Message already exists');
+
+    const entity = this.messageRepo.create(dto);
+
+    entity.supportIssue = await this.supportIssueRepo.findOneBy({ id: dto.supportIssue.id });
+    if (!entity.supportIssue) throw new NotFoundException('Support Issue not found');
```

Comment:
A bit more consistency in error messages would be good 😉 
- Support **i**ssue not found
- Support **m**essage already exists
- Support **i**ssue not found